routes/keyword.js

```diff
@@ -1,53 +1,81 @@
 const express = require('express')
-const Keyword = require('../models/KeywordModel')
+const Keyword = require('../models/KeywordModel.js')
 
 const router = express.Router()
 
 router.get('/top', (req, res) => {
   Keyword.find()
-    .lean()
-    .then((wordList) => {
-      const topWords = wordList.reduce((accumulator, word) => {
-        const wordSum = word.votes[word.votes.length - 1].sum
-        const topSum = accumulator.sum
-        if (wordSum > topSum) {
-          return {
-            words: [word],
-            sum: wordSum,
-          }
-        }
-        if (wordSum === topSum) {
-          accumulator.words.push(word)
-        }
-        return accumulator
-      }, {
-        sum: 0,
-        words: [],
-      })
-      res.send(topWords.words)
-    })
-    .catch((err) => {
-      res.status(500).send({
-        err,
-        givens: req.query,
-      })
-    })
+        .lean()
+        .then((wordList) => {
+          const topWords = wordList.reduce((accumulator, word) => {
+            const wordSum = word.votes[word.votes.length - 1].sum
+            const topSum = accumulator.sum
+            if (wordSum > topSum) {
+              return {
+                words: [word],
+                sum: wordSum,
+              }
+            }
+            if (wordSum === topSum) {
+              accumulator.words.push(word)
+            }
+            return accumulator
+          }, {
+            sum: 0,
+            words: [],
+          })
+          res.send(topWords.words)
+        })
+        .catch((err) => {
+          res.status(500).send({
+            err,
+            givens: req.query,
+          })
+        })
 })
 
 router.get('/:word', (req, res) => {
   const word = req.params.word
   Keyword.findOne({ word })
-    .lean()
-    .then((keyword) => {
-      if(!keyword) throw new Error('Word not found')
-      res.send([keyword])
-    })
-    .catch((err) => {
-      res.status(500).send({
-        err: err.toString(),
-        givens: req.params,
-      })
-    })
+        .lean()
+        .then((keyword) => {
+          if (!keyword) throw new Error('Word not found')
+          res.send([keyword])
+        })
+        .catch((err) => {
+          res.status(500).send({
+            err: err.toString(),
+            givens: req.params,
+          })
+        })
 })
 
-module.exports = router
+function incrementKeyword(word, amount) {
+  return new Promise((resolve) => {
+    Keyword.findOne({ word })
+            .then((keyword) => {
+              let oldSum = 0
+              if (!keyword) {
+                keyword = new Keyword({
```

Comment:
This looks out of place, let's move the logic of adding a new keyword to the same time as they are created with the article